PSMDB-1796 Release notes for 7.0.24-13 (#1056)
percona/psmdb-docs: docs/release_notes/7.0.24-13.md

# Percona Server for MongoDB 7.0.24-13 ({{date.7_0_24}})

[Installation](../install/index.md){.md-button}
[Upgrade from MongoDB Community](../install/upgrade-from-mongodb.md){.md-button}

Percona Server for MongoDB 7.0.24-13 is an enhanced, source-available, and highly-scalable database that is a
fully-compatible, drop-in replacement for MongoDB Community Edition.

> Percona Server for MongoDB 7.0.24-13 includes the improvements and bug fixes of [MongoDB 7.0.23 Community Edition](https://www.mongodb.com/docs/manual/release-notes/7.0/#7.0.23---aug-13--2025) and MongoDB 7.0.24 Community Edition](https://www.mongodb.com/docs/manual/release-notes/7.0/#7.0.24---upcoming). It supports protocols and drivers of both MongoDB Community 7.0.23 and MongoDB Community 7.0.24.

Reviewer: @nastena1606

> Percona Server for MongoDB 7.0.24-13 includes the improvements and bug fixes of [MongoDB 7.0.23 Community Edition](https://www.mongodb.com/docs/manual/release-notes/7.0/#7.0.23---aug-13--2025) and MongoDB 7.0.24 Community Edition](https://www.mongodb.com/docs/manual/release-notes/7.0/#7.0.24---upcoming). It supports protocols and drivers of both MongoDB Community 7.0.23 and MongoDB Community 7.0.24.

Author: ```suggestion
Percona Server for MongoDB 7.0.24-13 includes the improvements and bug fixes of [MongoDB 7.0.23 Community Edition](https://www.mongodb.com/docs/manual/release-notes/7.0/#7.0.23---aug-13--2025) and [MongoDB 7.0.24 Community Edition](https://www.mongodb.com/docs/manual/release-notes/7.0/#7.0.24---upcoming). It supports protocols and drivers of both MongoDB Community 7.0.23 and MongoDB Community 7.0.24.
```

## Release highlights

This release provides the following features and improvements:

### Enhance authentication security via a token-based authentication flow with OpenID Connect (OIDC)

Percona Server for MongoDB now supports OpenID Connect (OIDC) / OAuth 2.0, providing a secure and simple way to manage user authentication using identity and access tokens. This feature allows you to centralize user management with a single identity provider, enhance security by eliminating the need to store credentials in your database and reduce the risk of credential theft. You can leverage enhanced authentication techniques like Single Sign-On (SSO) and Multi-Factor Authentication (MFA), improving your overall security posture and user experience.

Learn more about OIDC connect in our [documentation](../oidc.md)

This feature is available in [Percona Server for MongoDB Pro](../psmdb-pro.md) out of the box. You can also receive it by building Percona Server from the source code.

### No vendor lock-in with OpenBao support for data-at-rest encryption

> Percona Server for MongoDB is already integrated with HashiCorp Vault for enterprise-grade data encryption at rest. Now you have the flexibility to use OpenBao, the open-source fork of Vault, offering the same level of security—without the constraints of vendor lock-in.

Reviewer: I'd rather prefer the first sentence is about OpenBao than HC Vault.
```suggestion
To give you more flexibility for enterprise-grade data encryption, Percona Server for MongoDB now supports OpenBao in addition to HashiCorp Vault. OpenBao is the open-source fork of Vault, providing the same powerful security while helping you avoid vendor lock-in.
```

> Percona Server for MongoDB is already integrated with HashiCorp Vault for enterprise-grade data encryption at rest. Now you now have the flexibility to use OpenBao, the open-source fork of Vault, offering the same level of security—without the constraints of vendor lock-in.

Author: ```suggestion
Percona Server for MongoDB is already integrated with HashiCorp Vault for enterprise-grade data encryption at rest. Now you have the flexibility to use OpenBao, the open-source fork of Vault, offering the same level of security—without the constraints of vendor lock-in.
```

* [SERVER-103741](https://jira.mongodb.org/browse/SERVER-103741) - Added new metrics to provide better insight into database profile collection activity.

* [WT-14391](https://jira.mongodb.org/browse/WT-14391) - Prevented potential data corruption by adding eviction requirements checks before accessing internal data stores.

* [SERVER-104771](https://jira.mongodb.org/browse/SERVER-104771) - Fixed the issue with initial sync failing when the source has TTL indexes on a capped collection by allowing it from nodes with such indexes.

* [WT-14653](https://jira.mongodb.org/browse/WT-14653) - Added reconciliation stats for tracking history store updates

* [SERVER-103505](https://jira.mongodb.org/browse/SERVER-103505) - Prevented crashes that could occur when internal table information was unexpectedly missing.

Find the full list of changes in the release notes for:

* [MongoDB 7.0.23 Community Edition](https://www.mongodb.com/docs/manual/release-notes/7.0/#7.0.23---aug-13--2025)

> * MongoDB 7.0.24 Community Edition](https://www.mongodb.com/docs/manual/release-notes/7.0/#7.0.24---upcoming).

Reviewer: @nastena1606
